### PR TITLE
Feat/emailpro state

### PR DESCRIPTION
**Pull request by marie-j**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/solarsteein`
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | none
| License          | BSD 3-Clause

## Description

Revamp state & url for emailpro/mxplan

### CODE REVIEW

File: packages/manager/modules/emailpro/src/account/account.routing.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+const state = {
+  url: '/account',
+  template,
+  controller: 'EmailProTabAccountsCtrl',
+};
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('email-pro.dashboard.account', {
+    ...clone(state),
```
Comment (marie-j) [inline packages/manager/modules/emailpro/src/account/account.routing.js:12]: Can't use `state` directly as ui-router performs some updates on the parameter

File: packages/manager/modules/emailpro/src/account/alias/emailpro-account-alias.controller.js [hunk 2]
```diff
@@ -1,107 +1,122 @@
...
 
-export default /* @ngInject */ ($scope, $stateParams, $translate, EmailPro) => {
-  $scope.aliasMaxLimit = EmailPro.aliasMaxLimit;
-  $scope.getAliasesParams = {
-    pageSize: null,
-    offset: null,
-  };
+export default class {
+  /* @ngInject */
+  constructor($scope, $stateParams, $translate, EmailPro, goToAccounts) {
```
Comment (marie-j) [inline packages/manager/modules/emailpro/src/account/alias/emailpro-account-alias.controller.js:5]: Turned into a class to be able to resolve, didn't clean the code up

File: packages/manager/modules/emailpro/src/account/emailpro-account.controller.js [hunk 3]
```diff
@@ -12,281 +12,308 @@ import {
...
-    $scope.showAccounts = true;
-    $scope.showAlias = false;
-    $scope.selectedAccount = null;
-    $scope.noDomainFlag = true;
-    $scope.newConfiguredAccount = [];
-
-    return $q
+export default class {
+  /* @ngInject */
+  constructor(
```
Comment (marie-j) [inline packages/manager/modules/emailpro/src/account/emailpro-account.controller.js:17]: Turned into a class to be able to resolve, didn't clean the code up

File: packages/manager/modules/emailpro/src/dashboard/emailpro.controller.js [hunk 4]
```diff
@@ -277,6 +294,12 @@ export default /* @ngInject */ function EmailProCtrl(
     $scope.messageDetails = null;
   };
 
+  $scope.configureEmailService = function configure() {
```
Comment (marie-j) [inline packages/manager/modules/emailpro/src/dashboard/emailpro.controller.js:297]: Moved from previous tabs controller

**Comment by ovh-cds** [comment]
CDS Report build#14370.0 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#14370.1 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#14370.2 ✘  
* 
  * Build ✘

**Comment by ovh-cds** [comment]
CDS Report build#14376.0 ✘  
* 
  * Build ✘

File: packages/manager/modules/emailpro/src/redirection/redirection.routing.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import template from './emailpro-redirection.html';
+
+const state = {
+  url: '/redirection',
+  template,
+  controller: 'EmailMXPlanEmailRedirectionCtrl',
+};
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('mxplan.dashboard.redirection', clone(state));
```
Comment (JeremyDec) [inline packages/manager/modules/emailpro/src/redirection/redirection.routing.js:11]: You need to clone it here, even if you declare it just once ?

Comment (marie-j) [reply packages/manager/modules/emailpro/src/redirection/redirection.routing.js:11]: Copy/Paste and forgot to clean it up

**Comment by antleblanc** [comment]
@marie-j Could you please resolve existing conflicts? In order to be able to merge it.

Thank you!

**Comment by marie-j** [comment]
> @marie-j Could you please resolve existing conflicts? In order to be able to merge it.
> 
> Thank you!

@antleblanc Done !

File: packages/manager/modules/emailpro/src/account/emailpro-account.controller.js [hunk 6]
```diff
@@ -12,281 +12,308 @@ import {
...
+      );
+    };
+
+    this.$scope.$on(this.EmailPro.events.accountsChanged, () => {
+      this.$scope.$broadcast('paginationServerSide.reload', 'accountsTable');
+    });
+
+    // this.$scope.$on('showAccounts', () => {
+    //   this.$scope.displayAccounts();
+    // });
```
[the remark this answers is not in the document]
Comment (antleblanc) [reply packages/manager/modules/emailpro/src/account/emailpro-account.controller.js:133]: Thanks for pointing out @JeremyDec.
Could be addressed in a separate Pull Request when leaving draft status from #3330.

**Review by antleblanc — APPROVED** [review]

**Pull request merged**